### PR TITLE
fix(dep): Fixes out of date dependency info

### PR DESCRIPTION
**Pull request by thomastaylor312**
The help text for `helm dep` contained out of date info about referring to a chart repo by name vs. URL.

### CODE REVIEW

**Comment by thomastaylor312** [comment]
@technosophos I didn't add this to 2.5.1 in case it is too late, but it would definitely fit in the fix release

File: cmd/helm/dependency.go [hunk 1]
```diff
@@ -59,8 +59,8 @@ The 'version' field should contain a semantic version or version range.
 
 The 'repository' URL should point to a Chart Repository. Helm expects that by
 appending '/index.yaml' to the URL, it should be able to retrieve the chart
-repository's index. Note: 'repository' cannot be a repository alias. It must be
-a URL.
+repository's index. Note: 'repository' can be an alias. The alias must be prefaced
```
Comment (seh) [inline cmd/helm/dependency.go]: - [ ] Instead of "be prefaced by," consider "start with" instead.

**Comment by technosophos** [comment]
I think this just needs to be rebased to make it go green again

**Comment by thomastaylor312** [comment]
Ok @technosophos will do and will address @seh's comment at the same time

File: cmd/helm/dependency.go (hunk 1 above)
Comment (thomastaylor312) [inline cmd/helm/dependency.go]: Done

File: cmd/helm/dependency.go [hunk 2]
```diff
@@ -59,8 +59,8 @@ The 'version' field should contain a semantic version or version range.
 
 The 'repository' URL should point to a Chart Repository. Helm expects that by
 appending '/index.yaml' to the URL, it should be able to retrieve the chart
-repository's index. Note: 'repository' cannot be a repository alias. It must be
-a URL.
+repository's index. Note: 'repository' can be an alias. The alias must start
+with 'alias:' or '@'.
```
Comment (seh) [inline cmd/helm/dependency.go]: I noticed that this whole block of documentation uses quotation marks inconsistently: here, single quotes, elsewhere double quotes for words larger than a single character. That's fodder for a later cross-cutting documentation scrub.

**Review by seh — APPROVED** [review]

File: cmd/helm/dependency.go (hunk 2 above)
Comment (thomastaylor312) [inline cmd/helm/dependency.go]: Yeah, that is true. I need to create a clean up task for the docs

**Comment by thomastaylor312** [comment]
Oh, failing because I didn't make the docs. One sec

**Pull request merged**